src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -72,33 +67,41 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
   };
 
   useEffect(() => {
-    const batchId = batchIds[0];
-    const fetchBatch = async () => {
-      const response = await getBatch(Number(batchId));
-      if (response.requestSucceeded && response.batch) {
-        setBatch(response.batch);
-        setRecord({
-          purpose: 'Out Plant',
-          facilityId: response.batch.facilityId,
-          withdrawnDate: getTodaysDateFormatted(),
-          batchWithdrawals: [
-            {
-              batchId: response.batch.id,
-              notReadyQuantityWithdrawn: 0,
-              readyQuantityWithdrawn: 0,
-            },
-          ],
-        });
-      } else {
-        snackbar.toastError(response.error);
-      }
+    const fetchBatches = () => {
+      const withdrawalBatches: NurseryWithdrawal[] = [];
+      batchIds.forEach(async (batchId) => {
+        const response = await getBatch(Number(batchId));
```

Comment:
This will be very slow. Best to do one search API with a values field that is a list of all batch ids.
Should this come from the parent component  as a prop `originalBatches` or here is ok? if other components need batches then should come from parent. I think the grid of selecting quantities form needs batches (filtered down by selected nursery).
We may need to maintain a local state of records that is a 1:1 mapping from original batch to a record.
every time we update this local record, we can update the parent record based on a filtered down list of records (based on from nursery).

Basically:
```
Props {
  batches: any [] // not typed since it comes from search
  records;
  setRecords;
};

const [localRecords, setLocalRecords] = useState<>([]);
useEffect() {
   populate localREcords with NurseryWithdrawal
}

// for every change in the form
onChangeXYZ( {
   setLocalRecords(new changes)
   if (fromNursery) { // set records in parent, filtered by from nursery
      setRecords(localRecords.filter(facilityId === fromNursery)
   }
}
```